cmd/up.go

```diff
@@ -80,9 +86,107 @@ func newUpCmd() *cobra.Command {
 				Debug:                debug,
 			}
 
-			s, err := requireStack(stack, true, opts.Display)
-			if err != nil {
-				return err
+			var c config.Map
+			var s backend.Stack
+			hasStack := false
+
+			if len(args) > 0 {
+				url := args[0]
+
+				if !workspace.IsTemplateURL(url) {
+					return errors.Errorf("%s is not a valid URL", url)
+				}
+
+				// Retrieve the template repo.
+				var repo workspace.TemplateRepository
+				if repo, err = workspace.RetrieveTemplates(url, false); err != nil {
+					return err
+				}
+
+				// List the templates from the repo.
+				var templates []workspace.Template
+				if templates, err = repo.Templates(); err != nil {
+					return err
+				}
+
+				// Make sure only a single template is found.
+				// Alternatively, we could consider prompting to choose one instead of failing.
+				if len(templates) != 1 {
+					return errors.Errorf("more than one app template found at %s", url)
```

Comment:
Good point. We've been referring to these long form as "installable cloud applications". I'll just use "applications" in the error here instead of "app template".